test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -98,12 +98,24 @@ class test_uct_sockaddr : public uct_test {
         m_entities.push_back(client);
 
         /* initiate the client's private data callback argument */
-        client->client_cb_arg = server->iface_attr().max_conn_priv;
+        client->max_conn_priv = server->iface_attr().max_conn_priv;
 
         UCS_TEST_MESSAGE << "Testing " << m_listen_addr
                          << " Interface: " << GetParam()->dev_name;
     }
 
+    static ssize_t client_iface_priv_data_cb(void *arg, const char *dev_name,
+                                             void *priv_data)
+    {
+        size_t *max_conn_priv = (size_t*)arg;
+        size_t priv_data_len;
+
+        priv_data_len = uct_test::entity::priv_data_do_pack(priv_data);
+        EXPECT_LE(priv_data_len, (*max_conn_priv));
```

Comment:
what if pack_cb returns error (<0)? IMO, it should be ssize_t check to make error print cleaner

Reply:
i disagree with this comment: client_iface_priv_data_cb is the pack_cb which may (potentially) return <0.
on the other hand, priv_data_do_pack() is a well defined internal function which we can decide always returns >=0 so it can return size_t
we never assign priv_data_do_pack to a function pointer

Reply:
makes sense

Reply:
In addition, if this function is always called from entity's scope, I would recommend to define this as private.